continuous_integration.md

```diff
@@ -6,147 +6,174 @@ done with the corresponding commands `deno test`, `deno lint` and `deno fmt`. In
+        experimental: [ false ]
+        include: 
+          - deno-version: canary
+            os: ubuntu-20.04
+            experimental: true
+```
+
+### Speeding up Deno pipelines
+
+#### Reducing repetition
+
+In cross-platform runs, certain steps of a pipeline do not need to run for each
+OS necessarily. For example, generating the same test coverage report on Linux,
+MacOS and Windows is a bit redundant. You can use the `if` conditional keyword
+of GitHub Actions in these cases. The example below shows how to run code
+coverage generation and upload steps only on the `ubuntu` (Linux) runner:
 
 ```yaml
 - name: Generate coverage report
-  if: ${{ matrix.os == 'ubuntu-latest' }}
+  if: matrix.os == 'ubuntu-20.04'
   run: deno coverage --lcov cov > cov.lcov
+
+- name: Upload coverage to Coveralls.io
+  if: matrix.os == 'ubuntu-20.04'
+  uses: coverallsapp/github-action@master # Any code coverage service can be used, Coveralls.io is used here as an example.
+  with:
+    github-token: ${{ secrets.GITHUB_TOKEN }} # Generated by GitHub.
+    path-to-lcov: cov.lcov
 ```
 
-The same applies to uploading coverage to a reporter like Coveralls, for
-example:
+#### Caching dependencies
+
+As a project grows in size, more and more dependencies tend to be included. Deno
+will download these dependencies during testing and if a workflow is run many
+times a day, this can become a time-consuming process. A common solution to
+speed things up is to cache dependencies so that they do not need to be
+downloaded anew.
+
+As you might have read elsewhere in this manual, Deno stores dependencies
+locally in a cache directory. In a pipeline the cache can be preserved between
+workflows by setting the `DENO_DIR` environment variable and adding a caching
+step to the workflow:
 
 ```yaml
-- name: Upload coverage report
-  if: ${{ matrix.os == 'ubuntu-latest' }}
-  uses: coverallsapp/github-action@master
-  with:
-    github-token: ${{ secrets.GITHUB_TOKEN }}
-    path-to-lcov: cov.lcov
+# Set DENO_DIR to an absolute or relative path on the runner.
+env:
+  DENO_DIR: my_cache_directory
+
+steps:
+  - name: Cache Deno dependencies 
+    uses: actions/cache@v2
+    with:
+      path: my_cache_directory
+      key: ${{ runner.os }}-deno-cache-key
+```
+
+At first, when this workflow runs the cache is still empty and commands like
+`deno test` will still have to download dependencies, but when the job succeeds
+the contents of `DENO_DIR` are saved and any subsequent runs can restore them
+from cache instead of re-downloading.
+
+There is still one issue with the above approach: the cache key name is
+hardcoded on each OS, meaning the same cache will be restored every time, even
+if one or more dependencies are updated. The solution is to use a lockfile and
+to generate a different key each time the cache need to be refreshed. You can
+find more information on using lockfiles in Deno
+[here](./linking_to_external_code/integrity_checking.md).
```

Comment:
I find this paragraph confusing; I don't really know if it's needed. You should have more or less the same deps on all OSes so maybe there's no need to key cache by os?

Also there should be information about cache-busting in case your `DENO_DIR` gets malformed for whatever reason

Reply:
You're right, the OS runner isn't really needed in the key name so I removed it :). I've rewritten things a bit and added a short example on how to generate a new cache after updating a lockfile, as well as some info on clearing the cache if it gets malformed. This also requires changing the key name (they are still working on a way to clear a cache [from the UI](https://github.com/actions/cache/issues/2#issuecomment-912209469)).